Supertext.Base.Dal.SqlServer.Tests/Utils/SqlResultConverterTest.cs

```diff
@@ -165,4 +165,180 @@ public void DecodeStructure_JsonEncodedFieldNames_Level2Json()
+    }
+
+    [TestMethod]
+    public void InterpretUtcDates_ObjectWithMultipleNestedObject_ConvertedInUtc()
+    {
+        var nestedObject1 = new TestEntity(Guid.NewGuid(),
+                                          "Nested 1",
+                                          DateTime.Now,
+                                          null);
+
+        var nestedObject2 = new TestEntity(Guid.NewGuid(),
+                                           "Nested 2",
+                                           DateTime.Now.AddDays(-5),
+                                           DateTime.Now,
+                                           nestedObject1);
+
+        var testData = new TestEntity(Guid.NewGuid(),
+                                      "Test",
+                                      DateTime.Now.AddDays(-2),
+                                      DateTime.Now,
+                                      nestedObject2);
+
+        var converter = new SqlResultConverter();
+        var results = converter.InterpretUtcDates(testData);
+
+        results.Id.Should().Be(testData.Id);
+        results.Name.Should().Be(testData.Name);
+        results.CreatedOn.Kind.Should().Be(DateTimeKind.Utc);
+        results.UpdatedOn!.Value.Kind.Should().Be(DateTimeKind.Utc);
+        results.NestedObject!.Id.Should().Be(nestedObject2.Id);
+        results.NestedObject.Name.Should().Be(nestedObject2.Name);
+        results.NestedObject.CreatedOn.Kind.Should().Be(DateTimeKind.Utc);
+        results.NestedObject.UpdatedOn!.Value.Kind.Should().Be(DateTimeKind.Utc);
+        results.NestedObject!.NestedObject!.Id.Should().Be(nestedObject1.Id);
+        results.NestedObject.NestedObject.Name.Should().Be(nestedObject1.Name);
+        results.NestedObject.NestedObject.CreatedOn.Kind.Should().Be(DateTimeKind.Utc);
+    }
+
+    [TestMethod]
+    public void InterpretUtcDates_ObjectWithCollection_ConvertedInUtc()
+    {
+        var item1 = new TestEntity(Guid.NewGuid(),
+                                           "Item 1",
+                                           DateTime.Now,
+                                           null);
+
+        var item2 = new TestEntity(Guid.NewGuid(),
+                                           "Item 2",
+                                           DateTime.Now.AddDays(-5),
+                                           DateTime.Now);
+
+        var testData = new TestEntity(Guid.NewGuid(),
+                                      "Test",
+                                      DateTime.Now.AddDays(-2),
+                                      DateTime.Now,
+                                      NestedCollection: [item1, item2]);
+
+        var converter = new SqlResultConverter();
+        var results = converter.InterpretUtcDates(testData);
+
+        results.Id.Should().Be(testData.Id);
+        results.Name.Should().Be(testData.Name);
+        results.CreatedOn.Kind.Should().Be(DateTimeKind.Utc);

```

Comment:
Again, leave only the `NestedCollection` assertions in this test; remove the `CreatedOn` and `UpdatedOn` assertions.